t/porting/libperl.t

```diff
@@ -216,32 +216,40 @@ sub nm_parse_gnu {
             if (/^[Rr] (\w+)$/) {
                 # R: read only (const)
                 $symbols->{data}{const}{$1}{$symbols->{o}}++;
+                push @{$symbols->{xref}{$1}}, "$symbols->{o}: $line";
             } elsif (/^r .+$/) {
                 # Skip local const (read only).
             } elsif (/^([Tti]) (\w+)(\..+)?$/) {
                 $symbols->{text}{$2}{$symbols->{o}}{$1}++;
+                push @{$symbols->{xref}{$2}}, "$symbols->{o}: $line";
             } elsif (/^C (\w+)$/) {
                 $symbols->{data}{common}{$1}{$symbols->{o}}++;
+                push @{$symbols->{xref}{$1}}, "$symbols->{o}: $line";
             } elsif (/^[BbSs] (\w+)(\.\d+)?$/) {
                 # Bb: uninitialized data (bss)
                 # Ss: uninitialized data "for small objects"
                 $symbols->{data}{bss}{$1}{$symbols->{o}}++;
+                push @{$symbols->{xref}{$1}}, "$symbols->{o}: $line";
             } elsif (/^D _LIB_VERSION$/) {
                 # Skip the _LIB_VERSION (not ours, probably libm)
             } elsif (/^[DdGg] (\w+)$/) {
                 # Dd: initialized data
                 # Gg: initialized "for small objects"
                 $symbols->{data}{data}{$1}{$symbols->{o}}++;
+                push @{$symbols->{xref}{$1}}, "$symbols->{o}: $line";
             } elsif (/^. \.?(\w+)$/) {
                 # Skip the unknown types.
                 print "# Unknown type: $line ($symbols->{o})\n";
             }
             return;
         } elsif (/^ {8}(?: {8})? U _?(\w+)$/) {
             my ($symbol) = $1;
+            push @{$symbols->{xref}{$symbol}}, "$symbols->{o}: $line";
             return if is_perlish_symbol($symbol);
             $symbols->{undef}{$symbol}{$symbols->{o}}++;
             return;
+        } elsif (/^\s+U _?(\w+)$/) {
```

Comment:
This looks really similar to the preceding pattern. (AFAICT the only difference is that it's more lenient about whitespace.) Should they really be treated differently?